src/instruction/instructions/shift_rotate/asrs.ts

```diff
@@ -0,0 +1,126 @@
+import { IMemory } from 'board/memory/interfaces'
+import { Registers } from 'board/registers'
+import { ILabelOffsets } from 'instruction/interfaces'
+import {
+  checkOptionCount,
+  create,
+  createImmediateBits,
+  createLowRegisterBits,
+  getBits,
+  isImmediate,
+  isOptionCountValid,
+  setBits
+} from 'instruction/opcode'
+import { Halfword, Word } from 'types/binary'
+import { BaseInstruction } from '../base'
+
+export class AsrsRegisterInstruction extends BaseInstruction {
+  public name: string = 'ASRS'
+  public pattern: string = '0100000100XXXXXX'
+  private rdnPattern: string = '0100000100000XXX'
+  private rmPattern: string = '0100000100XXX000'
+  private expectedOptionCount: number = 3
+
+  public encodeInstruction(options: string[], labels: ILabelOffsets): Halfword {
+    checkOptionCount(options, this.expectedOptionCount)
+    if (options[0] !== options[1])
```

Comment:
If you check equality of the arg0 and arg1 in `canEncodeInstruction()` this check here is a bit redundant. But nevertheless I think here the check is correct but it should be removed in the `canEncodeInstruction()` method.